elephas/spark_model.py

```diff
@@ -263,12 +263,14 @@ def _evaluate(model, optimizer, loss, custom_objects, metrics, kwargs, data_iter
             return agg_loss / number_of_samples
         else:
             # if we do have metrics, we want to return a list of [loss value, metric value] - to match the keras API
-            agg_loss, agg_metric, number_of_samples = results.\
-                map(lambda x: (x[2] * x[0], x[2] * x[1], x[2])).\
-                reduce(lambda x, y: (x[0] + y[0], x[1] + y[1], x[2] + y[2]))
+            mapping_function = lambda x: tuple(x[-1] * x[i] for i in range(len(x) - 1)) + (x[-1],)
+            reducing_function = lambda x, y: tuple(x[i] + y[i] for i in range(len(x)))
```

Comment:
Could use `zip` and just iterate over the variables in the lists directly but I couldn't come up with good variable names. :man_shrugging: